src/NHibernate/Type/TypeFactory.cs

```diff
@@ -386,6 +386,22 @@ private static TypeClassification GetTypeClassification(string typeName)
 		/// This method will return null if the name is not found in the basicNameMap.
 		/// </remarks>
 		public static IType Basic(string name)
+		{
+			return Basic(name, null);
+		}
+
+		/// <summary>
+		/// Given the name of a Hibernate type such as Decimal, Decimal(19,0),
+		/// Int32, or even NHibernate.Type.DecimalType, NHibernate.Type.DecimalType(19,0),
+		/// NHibernate.Type.Int32Type, then return an instance of NHibernate.Type.IType
+		/// </summary>
+		/// <param name="name">The name of the type.</param>
+		/// <param name="parameters">The parameters for the type, if any.</param>
+		/// <returns>The instance of the IType that the string represents.</returns>
+		/// <remarks>
+		/// This method will return null if the name is not found in the basicNameMap.
+		/// </remarks>
+		public static IType Basic(string name, IDictionary<string, string> parameters)
```

Comment:
Support of parameters was lacking in case of built-in types. I was not expecting this, since `DateType` is parameterized... Its parameter could then only work by mapping the type with its .Net fullname instead of using its alias.